### PR TITLE
Init new custom mailer class

### PR DESCRIPTION
**Pull request by pfefferle**
I forgot to load the mailer 🫣

And added some line breaks

Screenshot 2024-12-10 at 17 09 57
Screenshot 2024-12-10 at 17 10 03
Screenshot 2024-12-10 at 17 10 08
Screenshot 2024-12-10 at 17 10 12

Bonus: Direct-Messages

Fix #1012

### CODE REVIEW

File: includes/class-mailer.php [hunk 1]
```diff
@@ -50,10 +53,10 @@ public static function comment_notification_subject( $subject, $comment_id ) {
 			return $subject;
 		}
 
-		$post = get_post( $comment->comment_post_ID );
+		$post = \get_post( $comment->comment_post_ID );
 
 		/* translators: %1$s: Blog name, %2$s: Post title */
```
Comment (obenland) [inline includes/class-mailer.php]:
```suggestion
		/* translators: 1: Blog name, 2: Like or Repost, 3: Post title */
```

File: includes/class-mailer.php [hunk 2]
```diff
@@ -123,15 +126,60 @@ public static function new_follower( $notification ) {
...
 
-		/* translators: %1$s: Blog name, %2$s: Follower name */
+		/* translators: 1: Blog name, 2: Follower name */
 		$subject = \sprintf( \__( '[%1$s] Follower: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] );
-		/* translators: %1$s: Blog name, %2$s: Follower name */
-		$message = \sprintf( \__( 'New follower: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] ) . "\r\n";
+		/* translators: 1: Blog name, 2: Follower name */
+		$message = \sprintf( \__( 'New Follower: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] ) . "\r\n\r\n";
 		/* translators: %s: Follower URL */
 		$message .= \sprintf( \__( 'URL: %s', 'activitypub' ), $actor['url'] ) . "\r\n\r\n";
```
Comment (obenland) [inline includes/class-mailer.php]:
```suggestion
		$message .= \sprintf( \__( 'URL: %s', 'activitypub' ), esc_url( $actor['url'] ) ) . "\r\n\r\n";
```

Comment (obenland) [reply includes/class-mailer.php]: Everything in `$message` should be escaped, actually. I didn't check that earlier.

File: includes/class-mailer.php [hunk 3]
```diff
@@ -123,15 +126,60 @@ public static function new_follower( $notification ) {
...
+				return;
+			}
+
+			$email = $user->user_email;
+		}
+
+		/* translators: %1$s: Blog name, %2$s: Actor name */
+		$subject = \sprintf( \__( '[%1$s] Direct-Message from: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] );
+		/* translators: %1$s: Blog name, %2$s: Actor name */
+		$message = \sprintf( \__( 'New Direct-Message: %2$s', 'activitypub' ), get_option( 'blogname' ), \wp_strip_all_tags( $activity['object']['content'], 'allowed_comment_html' ) ) . "\r\n\r\n";
```
Comment (obenland) [inline includes/class-mailer.php]:
```suggestion
		/* translators: 1: Blog name, 2 Actor name */
		$subject = \sprintf( \__( '[%1$s] Direct Message from: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] );
		/* translators: 1: Blog name, 2: Actor name */
		$message = \sprintf( \__( 'New Direct Message: %2$s', 'activitypub' ), get_option( 'blogname' ), \wp_strip_all_tags( $activity['object']['content'], 'allowed_comment_html' ) ) . "\r\n\r\n";
```

File: includes/class-mailer.php [hunk 4]
```diff
@@ -123,15 +126,60 @@ public static function new_follower( $notification ) {
...
+		/* translators: %s: Actor name */
+		$message .= \sprintf( __( 'From: %s', 'activitypub' ), $actor['name'] ) . "\r\n";
+		/* translators: %s: Actor URL */
+		$message .= \sprintf( \__( 'URL: %s', 'activitypub' ), $actor['url'] ) . "\r\n\r\n";
```
Comment (obenland) [inline includes/class-mailer.php]: same here with escaping

**Review by obenland — CHANGES_REQUESTED** [review]

**Review by obenland — DISMISSED** [review]

File: includes/class-mailer.php [hunk 5]
```diff
@@ -111,26 +115,73 @@ public static function new_follower( $notification ) {
...
+		/* translators: 1: Blog name, 2 Actor name */
+		$subject = \sprintf( \esc_html__( '[%1$s] Direct Message from: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \esc_html( $actor['name'] ) );
+		/* translators: 1: Blog name, 2: Actor name */
+		$message = \sprintf( \esc_html__( 'New Direct Message: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \wp_strip_all_tags( $activity['object']['content'] ) ) . "\r\n\r\n";
+		/* translators: Actor name */
+		$message .= \sprintf( \esc_html__( 'From: %s', 'activitypub' ), \esc_html( $actor['name'] ) ) . "\r\n";
+		/* translators: Actor URL */
+		$message .= \sprintf( \esc_html__( 'URL: %s', 'activitypub' ), \esc_url( $actor['url'] ) ) . "\r\n\r\n";
 
 		\wp_mail( $email, $subject, $message );
```
Comment (pfefferle) [inline includes/class-mailer.php:186]: Should we support HTML mails for direct messages?

```suggestion
		$message = \sprintf( \esc_html__( 'New Direct Message: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \wp_kses( $activity['object']['content'] ) ) . "<br/><br/>";
		/* translators: Actor name */
		$message .= \sprintf( \esc_html__( 'From: %s', 'activitypub' ), \esc_html( $actor['name'] ) ) . "<br/>";
		/* translators: Actor URL */
		$message .= \sprintf( \esc_html__( 'URL: %s', 'activitypub' ), \esc_url( $actor['url'] ) ) . "<br/><br/>";

		\wp_mail( $email, $subject, $message, array( 'Content-Type: text/html' ) );
```

Comment (pfefferle) [reply includes/class-mailer.php:186]: this is not working code, just to show how it might be possible!

Comment (obenland) [reply includes/class-mailer.php:186]: I think it's fine as is for now.

**Review by obenland — APPROVED** [review]

**Pull request merged**